src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -226,19 +226,30 @@ private void executeColumnInsert(Connection conn, ArrayList<Object> columnData)
 		columnData.clear();
 	}
 
-	public static MysqlSavedSchema restore(Connection connection, MaxwellContext context) throws SQLException, IOException, InvalidSchemaError {
-		MysqlSavedSchema s = new MysqlSavedSchema(context.getServerID(), context.getCaseSensitivity());
+	public static MysqlSavedSchema restore(MaxwellContext context, BinlogPosition targetPosition) throws SQLException, InvalidSchemaError {
+		try ( Connection conn = context.getMaxwellConnection() ) {
+			Long schemaID = findSchema(conn, targetPosition, context.getServerID());
+			if (schemaID == null)
+				return null;
```

Comment:
yeah, I went over the code; your analysis is correct, this section of the code doesn't know anymore whether it's restoring a binlog position that was stored in the db or whether it's a brand new position.  I think that's.. mostly ok.  hard to say what to do if we get a binlog position but the schema goes missing.  I suppose it's as good as anything to recapture it.